src/libraries/System.Private.CoreLib/src/System/Array.Enumerators.cs

```diff
@@ -133,4 +140,39 @@ public T Current
 
         object? IEnumerator.Current => Current;
     }
+
+    internal abstract class GenericEmptyArrayEnumeratorBase
+    {
+#pragma warning disable CA1822 // https://github.com/dotnet/roslyn-analyzers/issues/5911
+        public bool MoveNext() => false;
+
+        public void Reset() { }
+
+        public void Dispose() { }
+#pragma warning restore CA1822
+    }
+
+    /// <summary>Provides an empty enumerator singleton.</summary>
+    /// <remarks>
+    /// If the consumer is using SZGenericArrayEnumerator elsewhere or is otherwise likely
+    /// to be using T[] elsewhere, SZGenericArrayEnumerator's singleton should be used.  Otherwise,
+    /// this singleton should be used, as it doesn't reference T[] in order to reduce footprint.
+    /// </remarks>
+    internal sealed class GenericEmptyArrayEnumerator<T> : GenericEmptyArrayEnumeratorBase, IEnumerator<T>
```

Comment:
```suggestion
    internal sealed class GenericEmptyEnumerator<T> : GenericEmptyEnumeratorBase, IEnumerator<T>
```
Nit: This enumerator has nothing to do with Arrays.

Reply:
Renamed.  I kept it in the same file, though.